src/http.ts

```diff
@@ -24,6 +26,14 @@ export interface HttpMessage {
 export interface HttpRequest extends HttpMessage {
   method: HttpMethod;
   url: string;
+  bodyToJson: <T>() => T;
+  replaceHeader: (header: string, value: string | string[]) => HttpRequest;
+  removeHeader: (header: string) => HttpRequest;
+  query: (name: string) => HttpQuery;
+  removeQuery: (name: string) => HttpRequest;
+  addQuery: (name: string, value: string | string[]) => HttpRequest;
+  replaceQuery: (name: string, value: string | string[]) => HttpRequest;
+  path: () => string;
```

Comment:
Should we change this to a property?

Reply:
I think that with the path we can take advantage later as a function to perform operations